pandas/io/common.py

```diff
@@ -460,10 +460,11 @@ def get_handle(
         handles.append(f)
 
     elif is_path:
-        if encoding:
+        is_binary = "b" in mode
+        if encoding and not is_binary:
             # Encoding
             f = open(path_or_buf, mode, encoding=encoding, errors=errors, newline="")
-        elif is_text:
+        elif is_text and not is_binary:
```

Comment:
isn’t this always true? eg it’s either is_text or is_binary? if not can u add a comment

Reply:
oh thanks, your right `is_text == not is_binary`

Reply:
well, they should be the inverse of each other based on the doc string but it seems that `to_csv` or whatever calls `get_handle` sets `is_text` to `True` even if `mode` contains a `'b'`. I will fix that in the function calling `get_handle`.

Reply:
It seems that `is_text` is not describing the format of the output but whether the stream that is saved is string or bytes (at least that's how `to_csv` is using it).

Reply:
ok that is confusing

can u rename something here

is_binary -> is_binary_mode
text -> text_encoding

?

Reply:
I renamed `is_binary` to `is_binary_mode` and changed the doc string of `is_text`